docs/DeveloperGuide.md

```diff
@@ -1,34 +1,617 @@
+    Figure 6 - Illustration of Storage Structure
+</p>
+
+The _Storage_ component is responsible for creating and loading modules and their respective data, as well as saving the data each time a change is made. It consists of two classes: `Loader` and `Writer`. At every moment, `Loader` only loads up to 1 module at a time and data for each module is stored separately. This is done to ensure fast loading and writing of files.
+
+#### Loader:
+
+* Loads the list of modules from the “Data” directory
+* Loads lesson and task data from the selected module’s “.txt” file
+
+#### Writer:
+
+* Creates all the directories required
+* Deletes files and directories
+* Creates the “.txt” file that saves the module’s lessons and tasks
+* Writes changes to the “.txt” file that saves the module’s lessons and tasks
+
+&nbsp;
+
+### Editor component
+
+**API**: `TextEditor.java`
+
+The _Editor_ component is responsible for opening the text editor to add or edit cheat-sheets/notes. It consists of two classes:
+
+#### Text Editor
+
+* Sets up the text editor
+* Loads existing file from Cheatsheet directory within a module for the edit cheat-sheet command
+* Flushes out the text from the editor when a different or new file is opened.
+* Adjusts the font size of the text within the editor
+* Detects mouse input to change font style and save the text
+* Saves the text from the text editor into a file
+
+#### ShortcutListener
+
+* Detects keyboard input for shortcuts
+
+&nbsp;
+
+### Common classes
+
+Classes that are used by multiple components:
+* `CommonMethods`: Stores methods that are used by multiple components
+* `Constants`: Stores constants
+* `Messages`: Stores strings that are printed by the `UI`
+* `DashboardCommands`: Enum of commands that can be used outside a module
+* `ModuleCommands`: Enum of commands that can be used inside a module
+
+&nbsp;
+
+[🡅 Back to Table of Contents](#table-of-contents)
+
+----
+
+## Implementation
+
+In this section, we highlight a few of the key features whose implementations are reflective of most of the commands available, as well as those that are more unique to GULIO.
+
+### Add Lesson
+
+The `AddLessonCommand` class is responsible for the creation and addition of a new `Lesson` object to the lesson list of a given module. The following sequence diagrams shows how a new `Lesson` is created and added to the lesson list.
+
+<p align="center">
+    <img width="973" src="developerGuideImages/addLesson1.png" alt="parse() Sequence Diagram"><br>
+    Figure 7 - parse() Sequence Diagram
+</p>
+
+The creation process is facilitated by the `Parser` class, which parses the appropriate arguments from the user input and initialises the `Lesson` object attributes with the parsed values.
+
+<p align="center">
+    <img width="973" src="developerGuideImages/addLessonCommand.png" alt="AddLessonCommand Constructor Sequence Diagram"><br>
+    Figure 8 - AddLessonCommand Constructor Sequence Diagram
```

Comment:
The diagrams are labelled as "Figures" but the elaboration does not refer the the labels